hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnection.java

```diff
@@ -246,14 +232,17 @@ private void logClose() {
     }
 
     protected void innerClose() throws IOException {
-        // todo: closing needs to be improved because we can end up with some channels not closed.
-        channel.close();
-
-        if (tpcChannels != null) {
-            for (Channel tpcChannel : tpcChannels) {
-                tpcChannel.close();
+        if (altoChannels != null) {
+            for (Channel altoChannel : altoChannels) {
+                try {
+                    altoChannel.close();
```

Comment:
is it possible to throw exception?

Reply:
It is part of the method signature, it might throw an IOException. So better be defensive here.